kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/Transaction.java

```diff
@@ -1,65 +1,61 @@
 package kin.sdk;
 
-import java.math.BigDecimal;
+import java.io.IOException;
 import kin.base.KeyPair;
+import kin.base.Memo;
+import kin.base.Network;
 
 public class Transaction {
 
-    private final KeyPair destination;
-    private final KeyPair source;
-    private final BigDecimal amount;
-    private final int fee;
-    private final String memo;
-
     /**
      * The transaction hash
      */
     private final TransactionId id;
 
-    private final kin.base.Transaction stellarTransaction;
-    private final WhitelistableTransaction whitelistableTransaction;
+    private final kin.base.Transaction baseTransaction;
 
-    Transaction(KeyPair destination, KeyPair source, BigDecimal amount, int fee, String memo,
-                TransactionId id, kin.base.Transaction stellarTransaction, WhitelistableTransaction whitelistableTransaction) {
-        this.destination = destination;
-        this.source = source;
-        this.amount = amount;
-        this.fee = fee;
-        this.memo = memo;
+    public Transaction(TransactionId id, kin.base.Transaction baseTransaction) {
         this.id = id;
-        this.stellarTransaction = stellarTransaction;
-        this.whitelistableTransaction = whitelistableTransaction;
+        this.baseTransaction = baseTransaction;
     }
 
-    public KeyPair getDestination() {
-        return destination;
+    public static Transaction decodeTransaction(String transactionEnvelope) throws IOException {
```

Comment:
add docs, what's the `IOException` probably related to streaming operations, maybe wrap it with a different clearer exception.

Reply:
added a 'DecodeTransactionException' object and throw it instead of the original one.
But left the cause and the message as before so we will know what exactly happened.
Is it fine? 
Do do you think we should put our own message and cause?